x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -275,27 +309,35 @@ void deleteSnapshots(Map<String, List<SnapshotInfo>> snapshotsToDelete,
 
     /**
      * Delete the given snapshot from the repository in blocking manner
+     *
+     * @param repo     The repository the snapshot is in
+     * @param snapshot The snapshot metadata
+     * @param listener {@link ActionListener#onResponse(Object)} is called if a {@link SnapshotHistoryItem} can be created representing a
+     *                  successful or failed deletion call. {@link ActionListener#onFailure(Exception)} is called only if interrupted.
      */
-    void deleteSnapshot(String slmPolicy, String repo, SnapshotId snapshot, SnapshotLifecycleStats slmStats) {
-        logger.info("[{}] snapshot retention deleting snapshot [{}]", repo, snapshot);
+    void deleteSnapshot(String slmPolicy, String repo, SnapshotInfo snapshot, SnapshotLifecycleStats slmStats,
```

Comment:
It doesn't appear that `SnapshotInfo` is used anywhere in this method now, I think it can go back to `SnapshotId` since that's the only part we want?